codespan-reporting/src/term/views.rs

```diff
@@ -303,21 +334,30 @@ where
                 // Check to see if we need to render any intermediate stuff
                 // before rendering the next line.
                 if let Some((next_line_index, _)) = lines.peek() {
-                    match next_line_index.checked_sub(line_index) {
+                    match next_line_index.checked_sub(*line_index) {
                         // Consecutive lines
                         Some(1) => {}
                         // One line between the current line and the next line
                         Some(2) => {
                             // Write a source line
                             let file_id = labeled_file.file_id;
+
+                            // This line was not intended to be rendered initially.
+                            // To render the line right, we have to get back the original labels.
+                            let labels = labeled_file
+                                .lines
+                                .iter()
+                                .find(|(index, _)| **index == line_index + 1)
+                                .map_or(&current_labels, |(_, line)| &line.multi_labels);
```

Comment:
I just noticed that `labeled_file.lines` is a `BTreeMap`, so we could probably do something like:

```rust
let labels = labeled_file
    .lines
    .get(line_index + 1)
    .map_or(&[], |(_, line)| &line.multi_labels);
```